### PR TITLE
🪟 Apply default sync mode logic to new streams

### LINKED ISSUE #16013: Apply default sync mode logic to new streams after refresh source schema
We currently handle the default sync mode logic in the FE. We though only apply this when the connection is not in edit mode, but created. That means if a user "refreshed source schema" of a connection and new streams are detected those new streams are not getting the default sync mode logic applied, but stay with whatever the config of the backend told us is default (which is not the correct one by the logic we try to go for).

We should make sure that the default sync mode logic in https://github.com/airbytehq/airbyte/blob/master/airbyte-webapp/src/views/Connection/ConnectionForm/calculateInitialCatalog.ts will apply as well for streams that are new after a "Refresh source schema".

### PR DESCRIPTION
**Pull request by teallarson**
## What
closes #16013

Previously, streams that were new after a refresh would default to the first alphabetically sorted sync mode they supported rather than using our `getOptimalSyncModes` method.

Now, we pass the stream descriptors of new streams into the `calculateInitialCatalog` method to ensure these streams are calculated.

## Testing
Create a connection with a source that you can add streams to (ie: a local Postgres)
Add stream(s)
Refresh schema
Note that their sync modes are calculated
Note that existing streams are not affected by (their configured sync modes remain)

### CODE REVIEW

File: airbyte-webapp/src/views/Connection/ConnectionForm/calculateInitialCatalog.ts [hunk 1]
```diff
@@ -119,18 +124,24 @@ const getOptimalSyncMode = (
...
-    const nodeStream = verifySourceDefinedProperties(verifySupportedSyncModes(nodeWithId), isNotCreateMode || false);
-
-    if (isNotCreateMode) {
-      return nodeStream;
-    }
-
-    return getOptimalSyncMode(verifyConfigCursorField(nodeStream), supportedDestinationSyncModes);
-  }),
-});
+  isNotCreateMode?: boolean,
```
Comment (josephkmh) [inline airbyte-webapp/src/views/Connection/ConnectionForm/calculateInitialCatalog.ts:127]: nitpick: maybe a matter of preference, but I find negative names for booleans harder to read - in this case I would prefer `!isCreateMode` to `isNotCreateMode`

File: airbyte-webapp/src/views/Connection/ConnectionForm/calculateInitialCatalog.ts [hunk 2]
```diff
@@ -119,18 +124,24 @@ const getOptimalSyncMode = (
...
+  newStreamDescriptors?: StreamDescriptor[]
+): SyncSchema => {
+  return {
+    streams: schema.streams.map<SyncSchemaStream>((apiNode, id) => {
+      const nodeWithId: SyncSchemaStream = { ...apiNode, id: id.toString() };
+      const nodeStream = verifySourceDefinedProperties(verifySupportedSyncModes(nodeWithId), isNotCreateMode || false);
+
+      // if the stream is new since a refresh, we want to verify cursor and get optimal sync modes
+      const matches = newStreamDescriptors?.filter(
+        (streamId) => streamId.name === nodeStream?.stream?.name && streamId.namespace === nodeStream.stream?.namespace
```
Comment (josephkmh) [inline airbyte-webapp/src/views/Connection/ConnectionForm/calculateInitialCatalog.ts:137]: Is `streamId` accurate? Maybe `newStream`?

File: airbyte-webapp/src/views/Connection/ConnectionForm/calculateInitialCatalog.ts [hunk 3]
```diff
@@ -119,18 +124,24 @@ const getOptimalSyncMode = (
...
+  isNotCreateMode?: boolean,
+  newStreamDescriptors?: StreamDescriptor[]
+): SyncSchema => {
+  return {
+    streams: schema.streams.map<SyncSchemaStream>((apiNode, id) => {
+      const nodeWithId: SyncSchemaStream = { ...apiNode, id: id.toString() };
+      const nodeStream = verifySourceDefinedProperties(verifySupportedSyncModes(nodeWithId), isNotCreateMode || false);
+
+      // if the stream is new since a refresh, we want to verify cursor and get optimal sync modes
+      const matches = newStreamDescriptors?.filter(
```
Comment (josephkmh) [inline airbyte-webapp/src/views/Connection/ConnectionForm/calculateInitialCatalog.ts:136]: If you're just interested whether any new stream descriptions match, maybe `array.some()` would be better than `array.filter()`? That would make the comparison on line 139 a little less verbose: `if (isNotCreateMode && !matches)`

**Review by joeykmh — APPROVED** [review]
Tested locally, works and logic looks sound 👍 just left a few thoughts on syntax

File: airbyte-webapp/src/views/Connection/ConnectionForm/calculateInitialCatalog.ts (hunk 2 above)
Comment (tealjulia) [reply airbyte-webapp/src/views/Connection/ConnectionForm/calculateInitialCatalog.ts:137]: This only contains the identifying information (name and namespace), not the entire stream object.

**Pull request merged**